shared/src/types/token.rs

```diff
@@ -63,6 +63,11 @@ impl Amount {
             micro: amount * 1_000_000,
         }
     }
+
+    /// return the f64 rappresentation
+    pub fn to_f64(&self) -> f64 {
+        self.micro as f64
```

Comment:
It's probably better to convert from the micro units
```suggestion
        self.micro as f64 / 1_000_000
```

Reply:
even better, I think it would be nicer to instead `impl From<Amount> for f64`. Then you can skip manually converting in places where you call `to_f64`, because the good_lp's API accepts e.g. `max<N: Into<f64>>(self, max: N)` (`Into` is automatically implemented from `From`)